src/infrasys/time_series_models.py

```diff
@@ -228,6 +229,242 @@ class SingleTimeSeriesScalingFactor(SingleTimeSeries):
+        single_time_series: SingleTimeSeries,
+        interval: timedelta,
+        horizon: timedelta,
+        name: str | None = None,
+        window_count: int | None = None,
+    ) -> "DeterministicSingleTimeSeries":
+        """Create a DeterministicSingleTimeSeries from a SingleTimeSeries.
+
+        Parameters
+        ----------
+        single_time_series
+            The SingleTimeSeries to wrap
+        name
+            Name assigned to the forecast (defaults to the same name as the SingleTimeSeries)
+        interval
+            Time between forecast windows (e.g., 1h for hourly forecasts)
+        horizon
+            Length of each forecast window (e.g., 6h for 6-hour forecasts)
+        window_count
+            Number of forecast windows to provide. If None, maximum possible windows will be used.
+
+        Returns
+        -------
+        DeterministicSingleTimeSeries
+
+        Notes
+        -----
+        This creates a perfect forecast by wrapping the original SingleTimeSeries
+        and accessing its data at the appropriate offsets for each window.
+        """
+        # Use defaults if parameters aren't provided
+        name = name if name is not None else single_time_series.name
+        resolution = single_time_series.resolution
+
+        # Calculate maximum possible window count if not provided
+        if window_count is None:
+            total_duration = single_time_series.length * resolution
+            usable_duration = total_duration - horizon
+            max_windows = (usable_duration // interval) + 1
+            window_count = int(max_windows)
+            if window_count < 1:
+                msg = (
+                    f"Cannot create any forecast windows with horizon={horizon} "
+                    f"from time series of length {total_duration}"
+                )
+                raise ValueError(msg)
+
+        # Ensure the base time series is long enough to support the requested forecast windows
+        horizon_steps = int(horizon / resolution)
+        interval_steps = int(interval / resolution)
+        total_steps_needed = interval_steps * (window_count - 1) + horizon_steps
+
+        if total_steps_needed > single_time_series.length:
+            msg = (
+                f"Base time series length ({single_time_series.length}) is insufficient "
+                f"for the requested forecast parameters (need {total_steps_needed} points)"
+            )
+            raise ValueError(msg)
+
+        # Create a 2D forecast matrix where each row is a forecast window
+        # and each column is a time step in the forecast horizon
+        if isinstance(single_time_series.data, pint.Quantity):
+            forecast_matrix = (
+                np.zeros((window_count, horizon_steps)) * single_time_series.data.units
+            )
+        else:
+            forecast_matrix = np.zeros((window_count, horizon_steps))
+
+        # Fill the forecast matrix with data from the original time series
```

Comment:
This is creating a new Deterministic, not sharing data from the original, as the docstrings indicate.